shell/generate-deployment-command.sh

```diff
@@ -0,0 +1,181 @@
+    # Source the .env.deployment file to load the variables
+    source .env.deployment
+
+    # Define the chain configurations
+    chains["arbitrum"]="$ARBITRUM_RPC_URL $ARBISCAN_API_KEY $ARBITRUM_CHAIN_ID $ARBITRUM_ADMIN $ARBITRUM_COMPTROLLER"
+    chains["arbitrum_sepolia"]="$ARBITRUM_SEPOLIA_RPC_URL $ARBISCAN_API_KEY $ARBITRUM_SEPOLIA_CHAIN_ID $ARBITRUM_SEPOLIA_ADMIN $ARBITRUM_SEPOLIA_COMPTROLLER"
+    chains["avalanche"]="$AVALANCHE_RPC_URL $SNOWTRACE_API_KEY $AVALANCHE_CHAIN_ID $AVALANCHE_ADMIN $AVALANCHE_COMPTROLLER"
+    chains["base"]="$BASE_RPC_URL $BASESCAN_API_KEY $BASE_CHAIN_ID $BASE_ADMIN $BASE_COMPTROLLER"
+    chains["bnb_smart_chain"]="$BSC_RPC_URL $BSCSCAN_API_KEY $BSC_CHAIN_ID $BSC_ADMIN $BSC_COMPTROLLER"
+    chains["gnosis"]="$GNOSIS_RPC_URL $GNOSISSCAN_API_KEY $GNOSIS_CHAIN_ID $GNOSIS_ADMIN $GNOSIS_COMPTROLLER"
+    chains["mainnet"]="$MAINNET_RPC_URL $ETHERSCAN_API_KEY $MAINNET_CHAIN_ID $MAINNET_ADMIN $MAINNET_COMPTROLLER"
+    chains["optimism"]="$OPTIMISM_RPC_URL $OPTIMISTIC_API_KEY $OPTIMISM_CHAIN_ID $OPTIMISM_ADMIN $OPTIMISM_COMPTROLLER"
+    chains["polygon"]="$POLYGON_RPC_URL $POLYGONSCAN_API_KEY $POLYGON_CHAIN_ID $POLYGON_ADMIN $POLYGON_COMPTROLLER"
+    chains["sepolia"]="$SEPOLIA_RPC_URL $ETHERSCAN_API_KEY $SEPOLIA_CHAIN_ID $SEPOLIA_ADMIN $SEPOLIA_COMPTROLLER"
+    chains["scroll"]="$SCROLL_RPC_URL $SCROLLSCAN_API_KEY $SCROLL_CHAIN_ID $SCROLL_ADMIN $SCROLL_COMPTROLLER"
+else
+    # Set bool to enter intaractive mode
+    INTERACTIVE=true
+
+    # Define the chain configurations
+    chains["arbitrum"]="$ARBITRUM_CHAIN_ID $ARBITRUM_COMPTROLLER"
+    chains["arbitrum_sepolia"]="$ARBITRUM_SEPOLIA_CHAIN_ID $ARBITRUM_SEPOLIA_COMPTROLLER"
+    chains["avalanche"]="$AVALANCHE_CHAIN_ID $AVALANCHE_COMPTROLLER"
+    chains["base"]="$BASE_CHAIN_ID $BASE_COMPTROLLER"
+    chains["bnb_smart_chain"]="$BSC_CHAIN_ID $BSC_COMPTROLLER"
+    chains["gnosis"]="$GNOSIS_CHAIN_ID $GNOSIS_COMPTROLLER"
+    chains["mainnet"]="$MAINNET_CHAIN_ID $MAINNET_COMPTROLLER"
+    chains["optimism"]="$OPTIMISM_CHAIN_ID $OPTIMISM_COMPTROLLER"
+    chains["polygon"]="$POLYGON_CHAIN_ID $POLYGON_COMPTROLLER"
+    chains["sepolia"]="$SEPOLIA_CHAIN_ID $SEPOLIA_COMPTROLLER"
+    chains["scroll"]="$SCROLL_CHAIN_ID $SCROLL_COMPTROLLER"
+fi
+
+# Check if the provided chain is defined
+if [[ ! -v "chains[$arg]" ]]; then
+    echo -e "\n${EC}Invalid chain: Chain configuration for '$arg' not found.${NC}"
+    exit 1
+fi
+
+# Check for arguments passed to the script
+if [[ $INTERACTIVE == true ]]; then
+    # load values from the terminal prompt
+    echo -e "${WC}.env.deployment missing, entering interactive mode.... ${NC}\n"
+    echo -e "1. Enter RPC URL: \c"
+    read rpc_url
+
+    echo -e "2. Enter etherscan API key: \c"
+    read api_key
+
+    echo -e "3. Enter admin address: \c"
+    read admin
+
+    echo -e "4. Enter max segment: \c"
+    read MAX_SEGMENT_COUNT
+
+    echo -e "5. Using deterministic script? (yes/no): \c"
```

Comment:
If the it is a deterministic script lets add one more input for the version:

`What version is this? (1.1.1/1.1.2): `

Reply:
Its hardcoded as `Version 1.1.1` in line 159 below. I am thinking if we should pick this information from `package.json` instead of hard-coding it. Wdyt?

Reply:
> do you think it would be possible to call the generate-deployment-command inside deploy-multi-chain so that we will not repeat so much

Yes. I can add an additional flag such as `--print` (or do you have any suggestion) which will print the command in terminal after simulating it. This also solves https://github.com/sablier-labs/v2-core/issues/749 where in case RPC endpoint throws error, we will know.

Reply:
> Its hardcoded as Version 1.1.1 in line 159 below. I am thinking if we should pick this information from package.json instead of hard-coding it. Wdyt?

it doesn't matter if it hardcoded, it can be removed and we should add :

```sh
echo -e "6. What version is this deployment? (1.1.1/1.1.2)"
read VERSION
```
 and then:  `"ChainID $chain_id, Version $VERSION"`.

 regarding to read it from `package.json`, i don't think it would be the best choice, because not every time that version would be used.

> Yes. I can add an additional flag such as --print 

great

Reply:
```bash
~ ./shell/deploy-multi-chain.sh --print --interactive --deterministic

Interactive mode activated. Provide details below:

1. Enter admin address: 0x6666cA940D2f4B65883b454b7Bc7EEB039f64fa3
2. Enter etherscan API key: etherscanapikey
3. Enter max segment count: 100

What version is this deployment? (1.1.1 / 1.1.2): 1.1.2
Compiling the contracts...

Deployment on sepolia started...
Enter RPC URL for sepolia network: https://sepolia.infura.io/v3/68967a55fb91488684a9c1000f321d7f
+ Deterministic address
+ Simulating on forked network
+ Printing command without action

forge script script/DeployDeterministicCore3.s.sol --rpc-url https://sepolia.infura.io/v3/68967a55fb91488684a9c1000f321d7f --sig 'run(string,address,address,uint256)' 'ChainID 11155111, Version 1.1.2' 0x6666cA940D2f4B65883b454b7Bc7EEB039f64fa3 0x2006d43E65e66C5FF20254836E63947FA8bAaD68 100 -vvv

End of it.
```